engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -279,25 +274,36 @@ private void updateAvailableFeatures() {
         }
     }
 
-    private void compileShaders(Set<ShaderProgramFeature> features) {
-        compileShader(GL20.GL_FRAGMENT_SHADER, features);
-        compileShader(GL20.GL_VERTEX_SHADER, features);
-    }
-
     private void compileAllShaderPermutations() {
```

Comment:
Can I beg for a javadoc or a block comment for this one of all methods?

I can see what it does, but it'd be good to have an explanation of what "features" are. Also, am I right in thinking that this method just makes sure that all computations can be correctly compiled but then discards all compiled shaders? It'd be good to explain that too in the javadoc.